Modules/FindCUDA.cmake

```diff
@@ -41,11 +41,26 @@
 #      nvcc in the generated source.  If you compile to PTX and then load the
 #      file yourself, you can mix bit sizes between device and host.
 #
+#   CUDA_ARCH_NAME (Default is "Auto" for regular builds and "Manual" for cross-builds)
+#   -- Controls the way CUDA_SELECT_NVCC_ARCH_FLAGS function selects CUDA compute architecture flags.
+#      If you are building for local install, "Auto" should work best as it determines local machine compute arch.
+#      If you are building for distribution, set to "Common" or "All" to cover common subsets of architectures.
+#      "Manual" makes CUDA_SELECT_NVCC_ARCH_FLAGS to get flags directly from ENV{CUDA_ARCH_BIN}
```

Comment:
Manual is till a bad name.  It should be something more descriptive about where the value comes from.  It could be ReadFrom_CUDA_ARCH_BIN or ReadFromEnvironment or something like that.  I'm not thrilled by having this controlled by an environment variable.  If the environment changes you wouldn't get any changes in your build until something caused a configure.  The configure could be triggered unintentionally (not mistakenly) and then stuff starts to change.  It is more clear if you want the build configuration to change you run ccmake or cmake again and change a cache (option) variable.  As it is if you want to change the value you have to change the environment variable and then manually run cmake to reconfigure.